src/utils/reductions.hpp

```diff
@@ -68,7 +68,11 @@ struct ReductionBase {
     // Store the communicator in a shared_ptr, so that
     // MPI_Comm_free is called, but only when the last
     // copy of this ReductionBase is destroyed.
-    pcomm = std::shared_ptr<MPI_Comm>(new MPI_Comm, MPI_Comm_disconnect);
+    pcomm = std::shared_ptr<MPI_Comm>(new MPI_Comm, [](MPI_Comm *comm) {
+      auto *temp = comm;
+      PARTHENON_MPI_CHECK(MPI_Comm_disconnect(comm));
+      delete temp;
```

Comment:
What's the rationale behind this logic?
Is is about making sure that the disconnect is successful?
We could also check if `comm == MPI_COMM_NULL` following the call.

Reply:
The rational is that the constructor of the shared pointer call `new MPI_Comm`, but the user defined destructor never called `delete` on that object (`MPI_Comm_disconnect` doesn't call `delete` I think). The sanitizer was complaining about this and changing to this logic fixed the sanitizer issue.